pixl_dcmd/src/pixl_dcmd/_dicom_helpers.py

```diff
@@ -35,3 +39,47 @@ def get_project_name_as_string(dataset: Dataset) -> str:
         logger.debug(f"String slug '{raw_slug}'")
         slug = raw_slug
     return slug
+
+
+class DicomValidator:
+    def __init__(self, edition: str = "current"):
+        self.edition = edition
+
+        # Default from dicom_validator but defining here to be explicit
+        standard_path = str(Path.home() / "dicom-validator")
+        edition_reader = EditionReader(standard_path)
+        destination = edition_reader.get_revision(self.edition, False)
+        json_path = Path(destination, "json")
+        self.dicom_info = EditionReader.load_dicom_info(json_path)
+
+    def validate_original(self, dataset: Dataset) -> None:
+        # Temporarily disable logging to avoid spamming the console
+        logging.disable(logging.ERROR)
+        self.original_errors = IODValidator(dataset, self.dicom_info).validate()
+        logging.disable(logging.NOTSET)
+
+    def validate_anonymised(self, dataset: Dataset) -> dict:
+        # Check that the original dataset has been validated
+        try:
+            orig_errors = self.original_errors
+        except AttributeError:
+            raise ValueError("Original dataset not yet validated")
+
+        # Temporarily disable logging to avoid spamming the console
+        logging.disable(logging.ERROR)
+        self.anon_errors = IODValidator(dataset, self.dicom_info).validate()
+        logging.disable(logging.NOTSET)
```

Comment:
interesting, this suggests that we should only allow logging from pixl, rather than logging from all imported modules